diff --git a/dist/zod/Log.js b/dist/zod/Log.js
index 917c16a77..0b400230e 100644
--- a/dist/zod/Log.js
+++ b/dist/zod/Log.js
@@ -4,7 +4,7 @@ import * as z_Hex from './Hex.js';
 import * as z_Number from './Number.js';
 import * as z_Uint from './Uint.js';
 import * as z from 'zod/mini';
-const topics = z.tuple([z_Hex.Hex], z_Hex.Hex);
+const topics = z.array(z_Hex.Hex);
 /** Log schema. */
 export const Log = z.object(fields(z_Uint.Uint, z_Number.Number));
 /** Encode-only log schema accepting numberish `toRpc` inputs. */
