diff --git a/dist/client/theme-default/index.js b/dist/client/theme-default/index.js
index a26a11093aa91df0b9996443fe1d480249e0619f..9ed3e6ba305f3be687624fb41366454809a73932 100644
--- a/dist/client/theme-default/index.js
+++ b/dist/client/theme-default/index.js
@@ -10,6 +10,7 @@ import './styles/components/vp-sponsor.css';
 import VPBadge from './components/VPBadge.vue';
 import Layout from './Layout.vue';
 import NotFound from './NotFound.vue';
+export { default as VPButton } from './components/VPButton.vue';
 export { default as VPHomeHero } from './components/VPHomeHero.vue';
 export { default as VPHomeFeatures } from './components/VPHomeFeatures.vue';
 export { default as VPHomeSponsors } from './components/VPHomeSponsors.vue';