diff --git a/dist/client/theme-default/index.js b/dist/client/theme-default/index.js
index 3b7ecabca5c5db4ca2b2d36762041741666bfef9..7c51f7d43e75d1e38789fe8e99b241774942252f 100644
--- a/dist/client/theme-default/index.js
+++ b/dist/client/theme-default/index.js
@@ -10,6 +10,7 @@ import './styles/components/vp-sponsor.css';
 import VPBadge from './components/VPBadge.vue';
 import Layout from './Layout.vue';
 import NotFound from './NotFound.vue';
+export { default as VPButton } from './components/VPButton.vue';
 export { default as VPHomeHero } from './components/VPHomeHero.vue';
 export { default as VPHomeFeatures } from './components/VPHomeFeatures.vue';
 export { default as VPHomeSponsors } from './components/VPHomeSponsors.vue';
diff --git a/jsconfig.json b/jsconfig.json
new file mode 100644
index 0000000000000000000000000000000000000000..9cffa9d0541a68d18ff74b0f53a8b4625f103e69
--- /dev/null
+++ b/jsconfig.json
@@ -0,0 +1,7 @@
+{
+  "typeAcquisition": {
+    "include": [
+      "wechat-miniprogram"
+    ]
+  }
+}