diff --git a/dist/internal/shiki-transformers.js b/dist/internal/shiki-transformers.js
--- a/dist/internal/shiki-transformers.js
+++ b/dist/internal/shiki-transformers.js
@@ -273,6 +273,10 @@ function checkOnlyTwoslasher(options) {
         const twoslashOptions = {
             ...options.twoslashOptions,
             ...executeOptions,
+            compilerOptions: {
+                ...options.twoslashOptions?.compilerOptions,
+                ...executeOptions?.compilerOptions,
+            },
             handbookOptions: {
                 ...options.twoslashOptions?.handbookOptions,
                 ...executeOptions?.handbookOptions,
diff --git a/dist/react/internal/Outline.js b/dist/react/internal/Outline.js
--- a/dist/react/internal/Outline.js
+++ b/dist/react/internal/Outline.js
@@ -102,7 +102,7 @@ function useOutlineItems(options) {
     return { items, activeId };
 }
 export function Outline(props) {
-    const { className, minLevel = 2, maxLevel: maxLevelProp = 3, footer: Footer } = props;
+    const { className, minLevel = 2, maxLevel: maxLevelProp = 4, footer: Footer } = props;
     const { frontmatter } = MdxPageContext.use();
     const { outline = true } = frontmatter ?? {};
     const maxLevel = typeof outline === 'number' ? outline + 1 : maxLevelProp;
diff --git a/src/internal/shiki-transformers.ts b/src/internal/shiki-transformers.ts
--- a/src/internal/shiki-transformers.ts
+++ b/src/internal/shiki-transformers.ts
@@ -407,6 +407,10 @@ function checkOnlyTwoslasher(options: {
     const twoslashOptions = {
       ...options.twoslashOptions,
       ...executeOptions,
+      compilerOptions: {
+        ...options.twoslashOptions?.compilerOptions,
+        ...executeOptions?.compilerOptions,
+      },
       handbookOptions: {
         ...options.twoslashOptions?.handbookOptions,
         ...executeOptions?.handbookOptions,
diff --git a/src/react/internal/Outline.tsx b/src/react/internal/Outline.tsx
--- a/src/react/internal/Outline.tsx
+++ b/src/react/internal/Outline.tsx
@@ -122,7 +122,7 @@ function useOutlineItems(options: { minLevel: number; maxLevel: number }) {
 }
 
 export function Outline(props: Outline.Props) {
-  const { className, minLevel = 2, maxLevel: maxLevelProp = 3, footer: Footer } = props
+  const { className, minLevel = 2, maxLevel: maxLevelProp = 4, footer: Footer } = props
 
   const { frontmatter } = MdxPageContext.use()
   const { outline = true } = frontmatter ?? {}
